[Grid] Separate out clearing of grid item overriding sizes from computeAspectRatioDependentAndBaselineItems

sammygill opened the pull request:
#### 156db6b0c1095bc94706507f6128b55baa0ab516
<pre>
[Grid] Separate out clearing of grid item overriding sizes from computeAspectRatioDependentAndBaselineItems
<a href="https://bugs.webkit.org/show_bug.cgi?id=289989">https://bugs.webkit.org/show_bug.cgi?id=289989</a>
<a href="https://rdar.apple.com/147345726">rdar://147345726</a>

Reviewed by Brandon Stewart.

One of the things that RenderGrid::computeAspectRatioDependentAndBaselineItems()
does currently is reset the overriding sizes of its grid items. This
needs to be done because overriding sizes are the mechanism used to set
the sizes of the grid item computed as a result of grid layout, so we
need to clear these values before we perform layout in order to avoid
using stale values.

However, I do not see a reason why this needs to be coupled with the
rest of the logic in computeAspectRatioDependentAndBaselineItems. Seeing
this code here is a bit confusing, and I think it would make more sense
to do this as early as possible before we perform any sort of grid
layout logic, so I think we should move this next to the couple of other
functions that are performing grid-specific tasks before layout.

* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::clearGridItemOverridingSizesBeforeLayout):
(WebCore::RenderGrid::layoutGrid):
(WebCore::RenderGrid::layoutMasonry):

Canonical link: <a href="https://commits.webkit.org/292343@main">https://commits.webkit.org/292343@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f82447a84b41f1645dfeada7e37177ef49e365d9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95665 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/15269 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/5185 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100716 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/46170 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/97709 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/15557 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23706 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72955 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/30210 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98668 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11664 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/86408 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/53288 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/11378 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/4123 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/45508 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/81560 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/4242 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102750 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22719 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16584 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81997 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22971 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/82424 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/81351 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20384 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25942 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/3394 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/16060 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22687 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27840 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/22346 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25822 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/24088 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/rendering/RenderGrid.cpp:
```diff
@@ -330,6 +325,18 @@ void RenderGrid::layoutBlock(RelayoutChildren relayoutChildren, LayoutUnit)
+    // clear any override set previously, so it doesn't interfere in current layout
+    // execution.
+    for (auto& gridItem : childrenOfType<RenderBox>(renderGrid)) {
+        if (gridItem.isOutOfFlowPositioned() || gridItem.isLegend())
```
We should investigate in the future if these are really needed.

sammygill committed:
[Grid] Separate out clearing of grid item overriding sizes from computeAspectRatioDependentAndBaselineItems
https://bugs.webkit.org/show_bug.cgi?id=289989
rdar://147345726

Reviewed by Brandon Stewart.

One of the things that RenderGrid::computeAspectRatioDependentAndBaselineItems()
does currently is reset the overriding sizes of its grid items. This
needs to be done because overriding sizes are the mechanism used to set
the sizes of the grid item computed as a result of grid layout, so we
need to clear these values before we perform layout in order to avoid
using stale values.

However, I do not see a reason why this needs to be coupled with the
rest of the logic in computeAspectRatioDependentAndBaselineItems. Seeing
this code here is a bit confusing, and I think it would make more sense
to do this as early as possible before we perform any sort of grid
layout logic, so I think we should move this next to the couple of other
functions that are performing grid-specific tasks before layout.

* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::clearGridItemOverridingSizesBeforeLayout):
(WebCore::RenderGrid::layoutGrid):
(WebCore::RenderGrid::layoutMasonry):

Canonical link: https://commits.webkit.org/292343@main
